src/commonControls/numberControl/NumberControlAPL.ts

```diff
@@ -0,0 +1,167 @@
+            dataSource: numberPadDataSourceGenerator(),
+            customHandlingFuncs: [],
+        },
+        requestChangedValue: {
+            document: numberPadDocumentGenerator(),
+            dataSource: numberPadDataSourceGenerator(),
+            customHandlingFuncs: [],
+        },
+    };
+
+    /**
+     * The APL dataSource to use when requesting a number value
+     *
+     */
+    export function numberPadDataSourceGenerator() {
+        return (act: RequestValueAct | RequestChangedValueAct) => {
+            return {
+                numPadData: {
+                    controlId: act.control.id,
+                    headerTitle: i18next.t('NUMBER_CONTROL_DEFAULT_APL_HEADER_TITLE'),
+                    isValidValue: (act.control as NumberControl).state.isValidValue,
+                    validationFailedMessage: (act.control as NumberControl).getAplValidationFailedMessage(),
+                },
+            };
+        };
+    }
+
+    /**
+     * The APL document to use when requesting a number value.
+     * For information on EditText, see:
+     * https://developer.amazon.com/en-US/docs/alexa/alexa-presentation-language/apl-edittext.html
+     *
+     */
+    export function numberPadDocumentGenerator() {
+        return {
+            type: 'APL',
+            version: '1.4',
+            import: [
+                {
+                    name: 'alexa-layouts',
+                    version: '1.2.0',
+                },
+            ],
+            styles: {
+                EditStyle: {
+                    values: [
+                        {
+                            borderWidth: 2,
+                            borderStrokeWidth: 1,
+                            borderColor: 'darkgrey',
+                            hintColor: 'grey',
+                            hintWeight: '200',
+                            fontSize: '20dp',
+                        },
+                        {
+                            when: '${state.focused}',
+                            borderColor: 'green',
+                            borderStrokeWidth: 2,
+                        },
+                    ],
+                },
+            },
+            mainTemplate: {
+                parameters: ['numPadData'],
+                items: {
+                    type: 'Container',
+                    width: '100vw',
+                    height: '100vh',
+                    direction: 'column',
+                    items: [
+                        {
+                            type: 'AlexaBackground',
+                        },
+                        {
+                            type: 'AlexaHeader',
+                            headerTitle: '${numPadData.headerTitle}',
+                            headerDivider: true,
+                            theme: 'dark',
+                        },
+                        {
+                            type: 'Container',
+                            height: '75%',
```

Comment:
when you want to consume all remaining space, use 'height:auto' and/or 'grow: 1'.  I don't know all the subtleties so please have a quick read of the docs for those settings and share what you find.

Reply:
According to [docs](https://developer.amazon.com/en-US/docs/alexa/alexa-presentation-language/apl-data-types.html#special-dimensions): 'height: auto' sets the height to the component's natural size (i.e. exactly what the height needs to be to fit the component's contents). 
[grow/shrink](https://developer.amazon.com/en-US/docs/alexa/alexa-presentation-language/apl-container.html#grow-shrink) change the component size proportionally (i.e. 'grow: 1' stretches to fill 100% of remaining container space) and only apply if the component is inside a container/sequence. So 'grow: 1' is what I want here, thanks!